### PR TITLE
Add support for aten.split.Tensor followed by prim.ListUnpack (DO NOT MERGE - for upstreaming)

### PR DESCRIPTION
**Pull request by mgehre-amd**
Supports aten.split when followed by prim.ListUnpack via a canonicalization on prim.ListUnpack.
This sequence of ops appears in `start_logits, end_logits = x.split(1, dim=-1)`

### CODE REVIEW

**Comment by lisaliu1** [comment]
The code looks good to me. But I am not sure if we should merge this PR to xilinx/torch-mlir/main? 
@ljfitz I remember our discussion about contributing to the torch-mlir is the following:
* always upstream the implementation to the llvm/torch-mlir
* create feature branch in xilinx/torch-mlir to include the implementation if we need the code urgently.
If the above flow is the one we are following, then this PR should be a draft PR, and the create a PR to merge into llvm/torch-mlir

**Comment by mgehre-amd** [comment]
> The code looks good to me. But I am not sure if we should merge this PR to xilinx/torch-mlir/main? @ljfitz I remember our discussion about contributing to the torch-mlir is the following:
> 
> * always upstream the implementation to the llvm/torch-mlir
> * create feature branch in xilinx/torch-mlir to include the implementation if we need the code urgently.
>   If the above flow is the one we are following, then this PR should be a draft PR, and the create a PR to merge into llvm/torch-mlir

I agree with you. I opened this PR here to get your feedback & approval before I open an PR against llvm/torch-mlir.
I will follow your suggesting and convert it into a draft PR.

**Review by lisaliu1 — APPROVED** [review]

**Comment by mgehre-amd** [comment]
TODO: rebase on top of https://github.com/llvm/torch-mlir/pull/2102/files and turn into recompose pattern

**Comment by mgehre-amd** [comment]
Too late, someone else opened a PR for this upstream
